### DISCUSSION TITLE
Performance for Inpaint

### DISCUSSION THREAD

**Discussion by Foixa** [davefojtik/RunPod-Fooocus-API#25] [General]
Hello

When using with inpaint, requests take about 16 seconds. Is there any way to lower this?

Also, although I did not change any settings, it increased from 11 seconds to 16 seconds after 1 day. Could there be a problem with the Runpod?

Thanks

**Comment by davefojtik** [comment]
Inpaint requires additional models to be loaded and steps to be done. So it makes sense to take longer than e.g. standard txt2img. Performance on runpod can vary based on worker availability and the load of the selected data centre. But also how frequently the workers are used and if they're cached with [Flashboot](https://blog.runpod.io/introducing-flashboot-1-second-serverless-cold-start/). If you haven't used the endpoint for a while, it's normal to see a longer cold start on the first request as the cache can be cleared or whole workers shifted.

It's a dynamic cluster shared with many users and many workloads. We can't expect execution time stability comparable to a dedicated server with models preloaded and computing only this one task 24/7. That's a price we have to pay for paying only for what we use.

Regarding the optimization - if you try to squeeze every drop out of this, you can experiment with [cmd flags](https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#all-cmd-flags) and try to lower precisions or change attentions. Just beware that it can affect the output quality. Also, try to play with params. For example, just by not using `Fooocus V2` in `style_selections` you eliminate one step and model to be loaded.
